### PR TITLE
chore: bumping go mod to ibc-go/v7

### PR DESCRIPTION
**Pull request by damiannolan**
## Description

closes: #2920 

### Commit Message / Changelog Entry

```bash
chore: bumping go mod to ibc-go/v7
```

see the [guidelines](https://github.com/cosmos/ibc-go/blob/main/CONTRIBUTING.md#commit-messages) for commit messages. (view raw markdown for examples)

---

Before we can merge this PR, please make sure that all the following items have been
checked off. If any of the checklist items are not applicable, please leave them but
write a little note why.

- [ ] Targeted PR against correct branch (see [CONTRIBUTING.md](https://github.com/cosmos/ibc-go/blob/master/CONTRIBUTING.md#pr-targeting)).
- [ ] Linked to Github issue with discussion and accepted design OR link to spec that describes this work.
- [ ] Code follows the [module structure standards](https://github.com/cosmos/cosmos-sdk/blob/main/docs/docs/building-modules/10-structure.md).
- [ ] Wrote unit and integration [tests](https://github.com/cosmos/ibc-go/blob/master/CONTRIBUTING.md#testing).
- [ ] Updated relevant documentation (`docs/`) or specification (`x/<module>/spec/`).
- [ ] Added relevant `godoc` [comments](https://blog.golang.org/godoc-documenting-go-code).
- [ ] Provide a [commit message](https://github.com/cosmos/ibc-go/blob/main/CONTRIBUTING.md#commit-messages) to be used for the changelog entry in the PR description for review.
- [ ] Re-reviewed `Files changed` in the Github PR explorer.
- [ ] Review `Codecov Report` in the comment section below once CI passes.

### CODE REVIEW

**Comment by damiannolan** [comment]
I think this will need to be merged before trying to update the `e2e` go mod to use v7. cc @chatton

File: modules/core/03-connection/types/msgs_test.go [hunk 1]
```diff
@@ -132,12 +132,12 @@ func (suite *MsgTestSuite) TestNewMsgConnectionOpenTry() {
 		msg     *types.MsgConnectionOpenTry
 		expPass bool
 	}{
-		{"non empty connection ID", &types.MsgConnectionOpenTry{"connection-0", "clienttotesta", any, counterparty, 500, []*types.Version{ibctesting.ConnectionVersion}, clientHeight, suite.proof, suite.proof, suite.proof, clientHeight, signer}, false},
+		{"non empty connection ID", &types.MsgConnectionOpenTry{"connection-0", "clienttotesta", any, counterparty, 500, []*types.Version{ibctesting.ConnectionVersion}, clientHeight, suite.proof, suite.proof, suite.proof, clientHeight, signer, nil}, false},
```
Comment (crodriguezvega) [inline modules/core/03-connection/types/msgs_test.go:135]: Just noticed this and can be done in a follow-up PR, but is there a reason why we cannot use `NewMsgConnectionOpenTry` as with the other test cases?

**Review by crodriguezvega — APPROVED** [review]
Thank you, @damiannolan! Left just one question.

File: modules/core/03-connection/types/msgs_test.go (hunk 1 above)
Comment (damiannolan) [reply modules/core/03-connection/types/msgs_test.go:135]: Yeah, the constructor func doesn't set `PreviousConnectionId` which is being set here to force a failure. That field is deprecated so it must be empty in `ValidateBasic`

**Review by chatton — APPROVED** [review]
LGTM!

**Pull request merged**